src/weather.cpp

```diff
@@ -459,60 +459,77 @@ void weather_effect::lightning( int intensity )
+void weather_effect::morale( int intensity, int bonus, int bonus_max, int duration, int decay_start,
+                             const std::string &morale_id_str,
+                             const std::string &morale_msg, int morale_msg_frequency, game_message_type message_type )
 {
-    if( calendar::once_every( time_duration::from_seconds( intensity ) ) && is_player_outside() ) {
-        if( g->u.primary_weapon().has_flag( json_flag_RAIN_PROTECT ) && !one_in( 3 ) ) {
-            add_msg( _( "Your %s protects you from the acidic drizzle." ), g->u.primary_weapon().tname() );
-        } else {
-            if( g->u.worn_with_flag( json_flag_RAINPROOF ) && !one_in( 4 ) ) {
-                add_msg( _( "Your clothing protects you from the acidic drizzle." ) );
-            } else {
-                bool has_helmet = false;
-                if( g->u.is_wearing_power_armor( &has_helmet ) && ( has_helmet || !one_in( 4 ) ) ) {
-                    add_msg( _( "Your power armor protects you from the acidic drizzle." ) );
-                } else {
-                    add_msg( m_warning, _( "The acid rain stings, but is mostly harmless for now…" ) );
-                    if( one_in( 10 ) && ( g->u.get_pain() < 10 ) ) {
-                        g->u.mod_pain( 1 );
-                    }
-                }
-            }
-        }
+    if( !( calendar::once_every( time_duration::from_seconds( intensity ) ) && is_player_outside() ) ) {
+        return;
+    }
+
+    static const morale_type morale_id( morale_id_str );
+    if( !morale_id.is_valid() ) {
+        debugmsg( "Invalid morale ID: %s", morale_id_str.c_str() );
+        return;
+    }
+
+    get_avatar().add_morale( morale_id, bonus, bonus_max, 1_seconds * duration, 1_seconds * decay_start,
+                             true );
+    if( one_in( morale_msg_frequency ) ) {
+        add_msg( message_type, _( morale_msg ) );
     }
 }
 
 /**
- * Acid rain.
- * Causes major pain. Damages non acid-proof mobs. Very wet (acid).
+ * Effect.
+ * Causes the player to feel a status effect.
  */
-void weather_effect::acid( int intensity )
+void weather_effect::effect( int intensity, int duration,
+                             std::string bodypart_string, int effect_intensity, const std::string &effect_id_str,
+                             const std::string &effect_msg, int effect_msg_frequency, game_message_type message_type,
+                             std::string precipitation_name, bool ignore_armor, int clothing_protection,
+                             int umbrella_protection )
 {
     if( !( calendar::once_every( time_duration::from_seconds( intensity ) ) && is_player_outside() ) ) {
         return;
     }
 
-    auto &you = get_avatar();
-    if( you.primary_weapon().has_flag( json_flag_RAIN_PROTECT ) && one_in( 4 ) ) {
-        return add_msg( _( "Your umbrella protects you from the acid rain." ) );
+    if( !ignore_armor ) {
+        auto &you = get_avatar();
```

Comment:
```suggestion
        const auto &you = get_avatar();
```
this section does not modify the `avatar`